### PR TITLE
update publish script to check FW binary

### PR DESCRIPTION
**Pull request by xiulipan**
check FW binary key with [sof_ri_info.py](https://github.com/thesofproject/sof/blob/master/tools/sof_ri_info/sof_ri_info.py) tool
TODO:
- [ ] check FW binary TAG 
- [ ] check LDC files binary TAG

Example to use:

```
./publish.sh -c -i ~/work/sof/release/1.6.1/production/
========================================================================
checking /home/pxl/work/sof/release/1.6.1/production/sof-apl.ri
1f f4 58 74 64 d4 ae 90 ... b5 c2 49 4e 2a 5f 47 c2 (APL Intel prod key)
sof-apl.ri get correct key
========================================================================
checking /home/pxl/work/sof/release/1.6.1/production/sof-cnl.ri
41 a0 3e 14 1e 7e 29 72 ... cd 1d fa fd e8 a0 ba 91 (CNL Intel prod key)
sof-cnl.ri get correct key
========================================================================
checking /home/pxl/work/sof/release/1.6.1/production/sof-icl.ri
63 df 54 e3 c1 e5 d9 d2 ... 35 f1 ec 4d 93 73 e6 c4 (ICL Intel prod key)
sof-icl.ri get correct key
========================================================================
checking /home/pxl/work/sof/release/1.6.1/production/sof-tgl.ri
d3 72 92 99 4e b9 cd 67 ... 5d 62 4a 80 96 31 f8 b5 (TGL Intel prod key)
sof-tgl.ri get correct key
========================================================================
checking /home/pxl/work/sof/release/1.6.1/production/sof-ehl.ri
b5 b0 e2 25 3d c7 54 10 ... ad 23 46 0c 9f fc 3e b9 (EHL Intel prod key)
sof-ehl.ri get correct key
========================================================================
All FW binary pass check!
```

### CODE REVIEW

**Comment by xiulipan** [comment]
@lgirdwood Please check if this kind of scripts to help to do a binary pre-check.
And if the TODO list cover all checks we need for FW binary

**Review by lgirdwood — COMMENTED** [review]
So this tool tells us the binary is signed by a key but does not verify the signature. rimage can now verify signatures on all CAVS versions using the public keys. i.e. this would make the script check the file data is also correct and not corrupted.
Is there anyway we can validate 
1) the version is correct i.e. v1.6.1 instead of v1.6.1-fd34de. i.e. we are using a correct tag.
2) The LDC file is correct for each platform.
3) The topologies are correct version.

File: publish.sh [hunk 1]
```diff
@@ -5,7 +5,10 @@
 # stop on most errors
 set -e
 
-RELEASE_PLATFORMS=(byt cht bdw apl cnl icl jsl tgl ehl)
+# Need to check and update the release platforms
+NO_SIGNED_PLATFORMS=(byt cht bdw)
```
[the remark this answers is not in the document]
Comment (dbaluta) [reply publish.sh:11]: @lgirdwood @xiulipan can you please add imx8 imx8x imx8m to unsigned platforms?

[the remark this answers is not in the document]
Comment (xiulipan) [reply publish.sh:11]: @dbaluta I think it maybe better to have a new variable here? The check part will also try to check if file exist and I think NXP and Intel binary will go into different branches.

I will have a new `-n` option for NXP binary release, please help to check update.

**Comment by xiulipan** [comment]
> So this tool tells us the binary is signed by a key but does not verify the signature. rimage can now verify signatures on all CAVS versions using the public keys. i.e. this would make the script check the file data is also correct and not corrupted.
> Is there anyway we can validate
> 
> 1. the version is correct i.e. v1.6.1 instead of v1.6.1-fd34de. i.e. we are using a correct tag.
> 2. The LDC file is correct for each platform.
> 3. The topologies are correct version.

@lgirdwood A summary for what we may also TODO:
1. try to use rimage to check the signature:
I will try to play with this, if the verify will need the origin key, it would be hard for us to do the check. So a simple **KEY** check should be enough, we can trust the FW is not corrupted by on-device test.
2. The version:
I will try to check HEX directly to try to figure out if there is any **MAGIC** word or **OFFSET** for us to locate the TAG string. Then the compare would be simple.
3. LDC file
It seems we have LDC hash, DEBUG ABI and FW VERSION check with logger, I will try to port that validation with a scripts to compare the RI and LDC binary directly.
4. topology
I do not think we have TAG inside topology, the only way to check would be ABI version. Do we need to add some TAG or SHA info into the tplg files

**Comment by lgirdwood** [comment]
> * try to use rimage to check the signature:
>   I will try to play with this, if the verify will need the origin key, it would be hard for us to do the check. So a simple **KEY** check should be enough, we can trust the FW is not corrupted by on-device test.

only the public key is needed to check here. 

> * The version:
>   I will try to check HEX directly to try to figure out if there is any **MAGIC** word or **OFFSET** for us to locate the TAG string. Then the compare would be simple.

ok, this might not be easy. there is no dedicated section or entry ion manifest. nothing common ....

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**